src/Features/VisualBasic/Portable/ConvertLinq/VisualBasicConvertLinqQueryToLinqMethodProvider.vb

```diff
@@ -0,0 +1,234 @@
+                    expression = ProcessQueryClause(expression, clause)
+                    If expression Is Nothing Then
+                        Return Nothing
+                    End If
+                Next
+
+                Return expression.WithTrailingTrivia(source.GetTrailingTrivia())
+            End Function
+
+            Private Function ProcessQueryClause(expression As ExpressionSyntax, queryClause As QueryClauseSyntax) As ExpressionSyntax
+                Select Case (queryClause.Kind())
+                    Case SyntaxKind.FromClause
+                        Return ProcessFromClause(expression, DirectCast(queryClause, FromClauseSyntax))
+                    Case SyntaxKind.DistinctClause
+                        Return CreateInvocationExpression(expression, NameOf(Enumerable.Distinct))
+                    Case SyntaxKind.WhereClause
+                        Return CreateInvocationExpression(expression, NameOf(Enumerable.Where), DirectCast(queryClause, WhereClauseSyntax).Condition)
+                    Case SyntaxKind.SkipClause
+                        Return CreateInvocationExpression(expression, NameOf(Enumerable.Skip), DirectCast(queryClause, PartitionClauseSyntax).Count)
+                    Case SyntaxKind.TakeClause
+                        Return CreateInvocationExpression(expression, NameOf(Enumerable.Take), DirectCast(queryClause, PartitionClauseSyntax).Count)
+                    Case SyntaxKind.OrderByClause
+                        Return ProcessOrderByClause(expression, DirectCast(queryClause, OrderByClauseSyntax))
+                    Case SyntaxKind.SelectClause
+                        Return ProcessSelectClause(expression, DirectCast(queryClause, SelectClauseSyntax))
+                    Case SyntaxKind.GroupJoinClause
+                        Return Nothing ' Too complicated for query method syntax.
+                    Case SyntaxKind.SimpleJoinClause
+                        Return Nothing ' Too complicated for query method syntax.
+                    Case SyntaxKind.LetClause
+                        Return Nothing ' Too complicated for query method syntax.
+                    Case SyntaxKind.AggregateClause
+                        Return Nothing ' Too complicated for query method syntax.
+                    Case SyntaxKind.GroupByClause
+                        Return Nothing ' Group by in query and in method are not 100% equivalent. Consdier support of some group by in the future.
+                    Case Else
+                        Return Nothing
+                End Select
+            End Function
+
+            Private Function ProcessFromClause(parentExpression As ExpressionSyntax, fromClause As FromClauseSyntax) As ExpressionSyntax
+                Dim expression = parentExpression
+                Dim variables = fromClause.Variables
+                If variables.Count > 1 Then
+                    Return Nothing ' Do not support more than 1 variable. It complicates query methods.
```

Comment:
>Return Nothing ' Do not support more than 1 variable. It complicates query methods. [](start = 20, length = 83)

I don't see why would we not handle the case of multiple variables, it is equivalent tor the sequence of Froms with a single variable. I.e. ```From a in AA, b in BB``` is equivalent to ```From a in AA From b in BB```. Just do the same transformation in a loop.